### PR TITLE
feat:co-90: show required overlay content only

### PR DESCRIPTION
**Pull request by arjun-at-roam**
## PR Area

- [x] Frontend

## Type of change

_What kind of change does this PR introduce?_

- [x] Feature

## Description

* Before this PR, we were showing all the contents returned. This PR will filter through the contents array and only show whats required to be shown.

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [x] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

#### Evidence screenshots

Before:
Australia -
image
NZ -
image

Now:
Australia -
image

NZ -
image

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: src/components/tabs/Tabs.stories.ts [hunk 1]
```diff
@@ -32,14 +32,28 @@ Default.args = {
           id: '18months',
```
Comment (arjun-at-roam) [inline src/components/tabs/Tabs.stories.ts:32]: Storybook updated to have a `key` prop, courtesy of changes in `src/models/Tabs.ts`👇

File: src/models/Tabs.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import ProductEnum from 'src/models/enums/ProductEnum'
 
 export interface ContentsProps {
+  key: string
```
Comment (arjun-at-roam) [inline src/models/Tabs.ts:4]: `key` is an easy way to refer to content in code. `name` and `value` are prone to change.

File: src/modules/DialogOverlay.vue [hunk 3]
```diff
@@ -70,6 +79,23 @@ export default defineComponent({
...
+        'productPrice',
+        'indicativeMinMonthly',
+        'indicativeMonthly',
+      ],
+      fieldBreakdownNz: [
+        'interestFreePeriod',
+        'establishmentFee',
+        'productPrice',
+        'indicativeMinMonthly',
+      ],
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue:96]: NZ and AU have a different lists of things they should show to the user

Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue:96]: NZ and AU have a different lists of things they should show to the user

File: src/modules/DialogOverlay.vue [hunk 4]
```diff
@@ -78,21 +104,63 @@ export default defineComponent({
...
-    getDataForProduct(): TabItemProps[] | undefined {
+    getDataForProduct(): TabItemProps[] {
       const productType = this.selectProductType()
       const productData = this.tabsData.find(
         product => product.productType === productType
       )
-      return productData?.productItems
+      if (!productData)
+        throw new Error(`No data found corresponding to ${productType}`)
+      return productData.productItems
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue:114]: did this to get rid of `undefined` return type

File: src/modules/DialogOverlay.vue [hunk 5]
```diff
@@ -78,21 +104,63 @@ export default defineComponent({
...
+        productContents.findIndex(item => item.key === 'productPrice') === -1
+      ) {
+        productContents.push({
+          key: 'productPrice',
+          name: 'Purchase Amount',
+          value: `$${Number(this.productPrice).toFixed(2)}`,
+        })
+      }
+      return productContents
+    },
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue:132]: Product price comes from a different source. this function adds `productPrice` to the list of contents.

File: src/modules/DialogOverlay.vue [hunk 6]
```diff
@@ -78,21 +104,63 @@ export default defineComponent({
...
+          this.fieldBreakdownAu.includes(item.key)
+        )
+        return this.orderContent(displayedContent, this.fieldBreakdownAu)
+      }
+      if (this.lang === LanguageCodeEnum.NewZealand) {
+        const displayedContent = allContent.filter(item =>
+          this.fieldBreakdownNz.includes(item.key)
+        )
+        return this.orderContent(displayedContent, this.fieldBreakdownNz)
+      }
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue]: filter out stuff that only needs to be shown to users and sort them

**Review by jpyepez-roam — APPROVED** [review]
LGTM!

File: src/modules/DialogOverlay.vue [hunk 7]
```diff
@@ -78,21 +104,63 @@ export default defineComponent({
...
     },
-    tabsContents(id: string): ContentsProps[] | undefined {
+    appendProductPrice(productContents: ContentsProps[]): ContentsProps[] {
+      if (
+        productContents.findIndex(item => item.key === 'productPrice') === -1
+      ) {
+        productContents.push({
+          key: 'productPrice',
+          name: 'Purchase Amount',
+          value: `$${Number(this.productPrice).toFixed(2)}`,
```
Comment (ghost) [inline src/modules/DialogOverlay.vue:128]: should this be `parseFloat` to be more explicit?

File: src/modules/DialogOverlay.vue (hunk 6 above)
Comment (ghost) [inline src/modules/DialogOverlay.vue]: This seems a bit WET. Can you make it more DRY by turning it into a helper function?

**Review by ghost — APPROVED** [review]
Looks so good! Added a couple of minor suggestions ✌️

File: src/modules/DialogOverlay.vue (hunk 7 above)
Comment (arjun-at-roam) [reply src/modules/DialogOverlay.vue:128]: parseFloat would require a `String`.
image

Though `this.productPrice` is already a `Number`, I'm reemphasizing it to be a Number by using `Number(this.productPrice)`.
Without this emphasis, I'll get an error 👇 
image

File: src/modules/DialogOverlay.vue (hunk 6 above)
Comment (arjun-at-roam) [reply src/modules/DialogOverlay.vue]: created a function (`filterContentByLang`) to remove complexity outside `tabsContents`

**Pull request merged**